soc: espressif: refactor clock and RTC subsystems

LucasTambor opened the pull request:
The RTC subsystem in Espressif's SOCs, among other tasks, is responsible for clock selection for CPU and for low power domain clocks such as RTC_SLOW and RTC_FAST.

This commit allows for proper clock source and rate selection for CPU, using the espressif,riscv and espressif,xtensa-lx6/7 bindings.

It also enables clock selection for RTC_FAST and RTC_SLOW, that impacts some peripherals, such as rtc_timer.

It adds also changes for making the clock control api tests generic for any board.
All the device subsys definitions were moved to its own folder according to the clock compatible.
If the clock's async feature is not supported by the target, the test is skipped instead of failed.

Finally, it add tests for the rtc clk subsystem itself.

LucasTambor committed:
manifest: update hal espressif

Add changes to comply with rtc refactor.

Signed-off-by: Lucas Tamborrino <lucas.tamborrino@espressif.com>

LucasTambor committed:
tests: boards: espressif: Add RTC CLK test

Add tests for the rtc clk subsystem.

Signed-off-by: Lucas Tamborrino <lucas.tamborrino@espressif.com>

LucasTambor committed:
soc: esp32xx: refactor clock and RTC subsystems

The RTC subsystem in espressif's SOCs, among other tasks
is responsible for clock selection for CPU and for low
power domain clocks such as RTC_SLOW and RTC_FAST.

This commit allows for proper clock source and rate
selection for CPU, using the espressif,riscv and
espressif,xtensa-lx6/7 bindings.

It also enables clock selection for RTC_FAST and RTC_SLOW,
that impacts some peripherals, such as rtc_timer.

Signed-off-by: Lucas Tamborrino <lucas.tamborrino@espressif.com>

LucasTambor committed:
tests: drivers: clock control api: Make test generic

Changes for making the clock control api tests generic for
any board.
All the device subsys definitions were moved to its own folder
according to the clock compatible.
Also if the clock's async feature is not supported by the target,
the test is skipped instead of failed.

Signed-off-by: Lucas Tamborrino <lucas.tamborrino@espressif.com>

LucasTambor committed:
tests: drivers: counter: support rtc timer

Add support for rtc timer node in the test.

Signed-off-by: Lucas Tamborrino <lucas.tamborrino@espressif.com>

decsny reviewed (dismissed):
all the default values of properties in the DT bindings need justifications for why they are the default value in the descriptions

LucasTambor commented:
> all the default values of properties in the DT bindings need justifications for why they are the default value in the descriptions

@decsny there is no need indeed for this default values, it can depend mostly on the application and power requirements. I removed the default values and added `required: true` so no one misses them.
